docs/design/rubygems.md

```diff
@@ -0,0 +1,528 @@
+# Design document for RubyGems/Bundler package manager
+
+Contents:
+
+1. [Ruby ecosystem overview](#i-ruby-ecosystem-overview)
+2. [Current implementation overview (Cachito)](#ii-overview-of-the-current-implementation-in-cachito)
+3. [Design for the Cachi2 implementation](#iii-design-for-the-implementation-in-cachi2)
+
+## I. Ruby ecosystem overview
+
+### Development prerequisites
+In order to execute the commands in the examples below, make sure you have the following packages installed in your
+environment:
+
+```bash
+sudo dnf install rubygems rubygems-bundler
```

Comment:
You are right, it should be `rubygem`

Reply:
Are we OK with being RPM-centric here? Do we want to at least mention valid OS package names for e.g. Deb/Ubuntu and Arch (2 of the Big 3 :D)

Reply:
I used `dnf` because it is used in the docs for example [README.md](https://github.com/containerbuildsystem/cachi2?tab=readme-ov-file#virtual-environment)

Reply:
I also wonder how much value these bring. Can we just add links to the official docs?